test-framework/junit5-component/src/main/java/io/quarkus/test/component/SkipInject.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.test.component;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Skip injection for the annotated test method parameter.
+ */
+@Retention(RUNTIME)
+@Target(PARAMETER)
```

Comment:
Would it make sense to allow this on methods and classes where we would skip injection for all method parameters and for all methods respectively?
I am thinking that in case there is two or more extensions fighting over parameter injection, annotating each and every parameter can get pretty verbose.

Alternatively, weld-junit uses the opposite approach where you can annotate a method/class with [`@ExplicitParameterInjection`](https://github.com/weld/weld-testing/blob/master/junit5/src/main/java/org/jboss/weld/junit5/ExplicitParamInjection.java) and you then only resolve parameters that have CDI qualifiers and skip all others.

Reply:
> Would it make sense to allow this on methods and classes where we would skip injection for all method parameters and for all methods respectively?
I am thinking that in case there is two or more extensions fighting over parameter injection, annotating each and every parameter can get pretty verbose.

Hm, I doubt that anyone would use more than several test method parameters. However, it might make sense to support at least method-level annotation. 

> Alternatively, weld-junit uses the opposite approach where you can annotate a method/class with [@ExplicitParameterInjection](https://github.com/weld/weld-testing/blob/master/junit5/src/main/java/org/jboss/weld/junit5/ExplicitParamInjection.java) and you then only resolve parameters that have CDI qualifiers and skip all others.

The downside of this approach is that you need to use `@Default` which is not very convenient..

Reply:
> Hm, I doubt that anyone would use more than several test method parameters. However, it might make sense to support at least method-level annotation.

Yea, that would be good enough :+1: 

> The downside of this approach is that you need to use `@Default` which is not very convenient..

I see what you mean and I agree. Although from my experience (based on weld-junit issues) you rarely need to use this to begin with :shrug: